### ISSUE TITLE
"Unable to Retrieve Chrome Version"

### ISSUE THREAD

**Issue by linusbobcat** [dmarmor/epichrome#23]
Whenever I start up my Mac and attempt to start any Epichrome app, I get a "Unable to Retrieve Chrome Version" dialog. 

This can be fixed by persistently trying to start the app which will eventually make it start up normally. Once it fixes itself, the fix is applied to every Epichrome app.

I was on version 2.1.4. If this was fixed in 2.1.5, how do I update all the exisiting Epichrome apps?

**Comment by linusbobcat** [comment]
This issue was fixed in 2.1.5. Sorry for creating the issue.

**Comment by dmarmor** [comment]
No worries, thanks for the feedback. Do keep an eye on this, because I didn't change the way Epichrome queries Chrome's version in 2.1.5, so not sure why it wasn't working in 2.1.4 (it should be pretty robust). Please let me know if you start getting this error again...

**Comment by linusbobcat** [comment]
With the new update, I'm getting this error. 

"New version of Google Chrome found (/Applications/Google Chrome.app: could not
find /Applications/Google Chrome.app.). Updating app..."

followed by

"Unable to update app to new version of Google Chrome. (Unable to retrieve
Chrome version.) It may or may not run properly."

The apps still work perfectly though.

Chrome and Epichrome apps are both Chrome Version 45.0.2454.85 unknown (64-bit)

**Comment by dmarmor** [comment]
Hey @linusbobcat,

It looks like something is getting messed up in the logic that retrieves Chrome's version number. I can't quite figure out how it could be doing quite what it's doing. Is your command-line environment non-standard in any way?

Also, is Chrome actually in /Applications/Google Chrome.app on your system? Epichrome should still work no matter where it is, but that info would help me to know.

One test that might shed some light would be if you ran this command from a terminal:

```bash
mdls -name kMDItemVersion "/Applications/Google Chrome.app"
```
(use the path to Chrome on your system if it's not in /Applications)

And let me know what it returns? It'd be a start, at least. I'm reopening the issue until we know what's going on.

**Comment by linusbobcat** [comment]
I run the fish shell as my command line shell which I installed using homebrew. My command line environment isn't stock at all. I have Google Chrome installed on /Applications. The command strangely enough outputs:

```
kMDItemVersion = "45.0.2454.85"
```

which is consistent to the version for Epichrome apps and Chrome itself. I also have Google Chrome Canary installed if that changes anything.

**Comment by dmarmor** [comment]
Hmm. Well, offhand, I can't figure out what's going on here. Canary shouldn't be interfering. I've tracked the problem down to mdls. For some reason, when mdls runs in Epichrome it's as if Google Chrome.app is not in /Applications. But when you ran it last night from your command line, it worked perfectly, which I can't understand.

The only thing I can think to try would be if it's possible on your system to create a new user account, log in from that (which would presumably have a stock command-line environment) and see if it works when you run it from there. I know that's a pain to do, but it might tell us if it's something in your particular account environment.

Otherwise, I'm at a loss here...

**Comment by dmarmor** [comment]
Hey! Is it possible you have Spotlight indexing (or Spotlight itself) turned off for the root volume? I've just released [version 2.1.6](https://github.com/dmarmor/epichrome/releases), which I *think* should solve your problem.

Another user submitted an issue ([#28](https://github.com/dmarmor/epichrome/issues/28)) and the problem turned out to be that they had Spotlight disabled. When I disabled it on my system to try to fix the problem, I started getting the same errors you reported! So anyway, give 2.1.6 a whirl and let me know if it helps.

I'm going to optimistically close the issue, but feel free to re-open if you still have trouble.

**Comment by linusbobcat** [comment]
Yes, I did turn off Spotlight on the GUI side. I’ve updated all the Epichrome apps, rebooted, and the issue disappeared. Thanks for the swift response!
